src/runtime/manifest.ts

```diff
@@ -0,0 +1,85 @@
+// The MIT License (MIT)
+//
+// Copyright (c) 2021 Firebase
+//
+// Permission is hereby granted, free of charge, to any person obtaining a copy
+// of this software and associated documentation files (the "Software"), to deal
+// in the Software without restriction, including without limitation the rights
+// to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+// copies of the Software, and to permit persons to whom the Software is
+// furnished to do so, subject to the following conditions:
+//
+// The above copyright notice and this permission notice shall be included in
+// all copies or substantial portions of the Software.
+//
+// THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+// IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+// FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+// AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+// LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+// OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+// SOFTWARE.
+/**
+ * @internal
+ * An definition of a function as appears in the Manifest.
+ */
+export interface ManifestEndpoint {
+  entryPoint?: string;
+  region?: string[];
+  platform?: string;
+  availableMemoryMb?: number;
+  maxInstances?: number;
+  minInstances?: number;
+  concurrency?: number;
+  serviceAccountEmail?: string;
+  timeoutSeconds?: number;
+  vpc?: {
+    connector: string;
+    egressSettings?: string;
+  };
+  labels?: Record<string, string>;
+  ingressSettings?: string;
+  environmentVariables?: Record<string, string>;
+
+  httpsTrigger?: {
+    invoker?: string[];
+  };
+
+  callableTrigger?: {};
+
+  eventTrigger?: {
+    eventFilters: Record<string, string>;
```

Comment:
There was some discussion about how event filters can also take an operator field. Have we decided how we're going to handle that?

Reply:
Nope. I think we need to have a separate design/discussion, but since this manifest is still in alpha and not documented for our customers, we can improve it as we go.